From 21915a85659e1b98960cba9fac777bc2f7cee377 Mon Sep 17 00:00:00 2001
From: fuqiang <qiang.fu@spacemit.com>
Date: Thu, 14 Mar 2024 16:33:37 +0800
Subject: [PATCH 0484/1204] v2d : mv v2d from
 drivers/media/platform/spacemit/v2d to drivers/soc/spacemit/v2d

1.drivers/media/platform/spacemit is for driver based v4l2 framework
2.v2d driver is not based v4l2 framework

Change-Id: I189660075404faf0539e8183a9f77dec8482c957
Signed-off-by: fuqiang <qiang.fu@spacemit.com>
---
 drivers/media/platform/spacemit/Kconfig                   | 1 -
 drivers/media/platform/spacemit/Makefile                  | 1 -
 drivers/soc/spacemit/Kconfig                              | 1 +
 drivers/soc/spacemit/Makefile                             | 1 +
 drivers/{media/platform => soc}/spacemit/v2d/Kconfig      | 0
 drivers/{media/platform => soc}/spacemit/v2d/Makefile     | 0
 drivers/{media/platform => soc}/spacemit/v2d/csc_matrix.h | 0
 drivers/{media/platform => soc}/spacemit/v2d/v2d_drv.c    | 0
 drivers/{media/platform => soc}/spacemit/v2d/v2d_drv.h    | 0
 drivers/{media/platform => soc}/spacemit/v2d/v2d_hw.c     | 0
 drivers/{media/platform => soc}/spacemit/v2d/v2d_iommu.c  | 0
 drivers/{media/platform => soc}/spacemit/v2d/v2d_priv.h   | 0
 drivers/{media/platform => soc}/spacemit/v2d/v2d_reg.h    | 0
 13 files changed, 2 insertions(+), 2 deletions(-)
 rename drivers/{media/platform => soc}/spacemit/v2d/Kconfig (100%)
 rename drivers/{media/platform => soc}/spacemit/v2d/Makefile (100%)
 rename drivers/{media/platform => soc}/spacemit/v2d/csc_matrix.h (100%)
 rename drivers/{media/platform => soc}/spacemit/v2d/v2d_drv.c (100%)
 rename drivers/{media/platform => soc}/spacemit/v2d/v2d_drv.h (100%)
 rename drivers/{media/platform => soc}/spacemit/v2d/v2d_hw.c (100%)
 rename drivers/{media/platform => soc}/spacemit/v2d/v2d_iommu.c (100%)
 rename drivers/{media/platform => soc}/spacemit/v2d/v2d_priv.h (100%)
 rename drivers/{media/platform => soc}/spacemit/v2d/v2d_reg.h (100%)

diff --git a/drivers/media/platform/spacemit/Kconfig b/drivers/media/platform/spacemit/Kconfig
index a31a599b320c..5e602f78d651 100644
--- a/drivers/media/platform/spacemit/Kconfig
+++ b/drivers/media/platform/spacemit/Kconfig
@@ -3,5 +3,4 @@
 comment "Spacemit media platform drivers"
 
 source "drivers/media/platform/spacemit/vpu_k1x/Kconfig"
-source "drivers/media/platform/spacemit/v2d/Kconfig"
 source "drivers/media/platform/spacemit/camera/Kconfig"
diff --git a/drivers/media/platform/spacemit/Makefile b/drivers/media/platform/spacemit/Makefile
index cc22a9f000eb..16c63023baba 100644
--- a/drivers/media/platform/spacemit/Makefile
+++ b/drivers/media/platform/spacemit/Makefile
@@ -1,4 +1,3 @@
 # SPDX-License-Identifier: GPL-2.0-only
 obj-y += vpu_k1x/
-obj-y += v2d/
 obj-y += camera/
diff --git a/drivers/soc/spacemit/Kconfig b/drivers/soc/spacemit/Kconfig
index cb6f565ab3ee..88612802f0eb 100644
--- a/drivers/soc/spacemit/Kconfig
+++ b/drivers/soc/spacemit/Kconfig
@@ -29,4 +29,5 @@ config SPACEMI_K1X_DMA_RANGE
           effective
 
 source "drivers/soc/spacemit/jpu/Kconfig"
+source "drivers/soc/spacemit/v2d/Kconfig"
 source "drivers/soc/spacemit/spacemit-rf/Kconfig"
diff --git a/drivers/soc/spacemit/Makefile b/drivers/soc/spacemit/Makefile
index 39afd7a33819..5d865c275a69 100644
--- a/drivers/soc/spacemit/Makefile
+++ b/drivers/soc/spacemit/Makefile
@@ -1,5 +1,6 @@
 obj-$(CONFIG_SPACEMIT_PM_DOMAINS) += pm_domain/
 obj-$(CONFIG_CHIP_MEDIA_JPU)    += jpu/
+obj-$(CONFIG_SPACEMIT_V2D)    += v2d/
 obj-$(CONFIG_SPACEMIT_RFKILL)   += spacemit-rf/
 obj-$(CONFIG_SPACEMIT_REBOOT_CONTROL)	+= spacemit_reboot.o
 obj-$(CONFIG_SPACEMI_K1X_DMA_RANGE) += k1x-dma-range.o
diff --git a/drivers/media/platform/spacemit/v2d/Kconfig b/drivers/soc/spacemit/v2d/Kconfig
similarity index 100%
rename from drivers/media/platform/spacemit/v2d/Kconfig
rename to drivers/soc/spacemit/v2d/Kconfig
diff --git a/drivers/media/platform/spacemit/v2d/Makefile b/drivers/soc/spacemit/v2d/Makefile
similarity index 100%
rename from drivers/media/platform/spacemit/v2d/Makefile
rename to drivers/soc/spacemit/v2d/Makefile
diff --git a/drivers/media/platform/spacemit/v2d/csc_matrix.h b/drivers/soc/spacemit/v2d/csc_matrix.h
similarity index 100%
rename from drivers/media/platform/spacemit/v2d/csc_matrix.h
rename to drivers/soc/spacemit/v2d/csc_matrix.h
diff --git a/drivers/media/platform/spacemit/v2d/v2d_drv.c b/drivers/soc/spacemit/v2d/v2d_drv.c
similarity index 100%
rename from drivers/media/platform/spacemit/v2d/v2d_drv.c
rename to drivers/soc/spacemit/v2d/v2d_drv.c
diff --git a/drivers/media/platform/spacemit/v2d/v2d_drv.h b/drivers/soc/spacemit/v2d/v2d_drv.h
similarity index 100%
rename from drivers/media/platform/spacemit/v2d/v2d_drv.h
rename to drivers/soc/spacemit/v2d/v2d_drv.h
diff --git a/drivers/media/platform/spacemit/v2d/v2d_hw.c b/drivers/soc/spacemit/v2d/v2d_hw.c
similarity index 100%
rename from drivers/media/platform/spacemit/v2d/v2d_hw.c
rename to drivers/soc/spacemit/v2d/v2d_hw.c
diff --git a/drivers/media/platform/spacemit/v2d/v2d_iommu.c b/drivers/soc/spacemit/v2d/v2d_iommu.c
similarity index 100%
rename from drivers/media/platform/spacemit/v2d/v2d_iommu.c
rename to drivers/soc/spacemit/v2d/v2d_iommu.c
diff --git a/drivers/media/platform/spacemit/v2d/v2d_priv.h b/drivers/soc/spacemit/v2d/v2d_priv.h
similarity index 100%
rename from drivers/media/platform/spacemit/v2d/v2d_priv.h
rename to drivers/soc/spacemit/v2d/v2d_priv.h
diff --git a/drivers/media/platform/spacemit/v2d/v2d_reg.h b/drivers/soc/spacemit/v2d/v2d_reg.h
similarity index 100%
rename from drivers/media/platform/spacemit/v2d/v2d_reg.h
rename to drivers/soc/spacemit/v2d/v2d_reg.h
-- 
2.47.0

